atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -8,315 +8,63 @@
-                               value_nosubst, pattern, bool(match))
-                if match:
-                    break
-
-            if match:
-                attrs = parser.envs
-                key = match.groups()[0]
-
-        return attrs, key
-
-    @staticmethod
-    def get_next_release(current_release):
-        """
-        Calculate the incremented value.
-
-        :param current_release: str, current value
-        :return: str, incremented value
-        """
-
-        try:
-            return str(int(current_release) + 1)
-        except ValueError:
-            isdigit = type(current_release).isdigit
-            first_nondigit = [isdigit(x) for x in current_release].index(False)
-            next_int = str(int(current_release[:first_nondigit]) + 1)
-            return next_int + current_release[first_nondigit:]
-
-    def bump(self, repo, remote):
-        """
-        Push a commit with an incremented release value.
-
-        :param repo: str, GitRepo instance
-        :param remote: str, git remote to push to
-        """
-
-        # Set up configuration
-        repo.git(['config', 'push.default', 'simple'])
-        repo.git(['config', 'user.email', self.committer_email])
-        repo.git(['config', 'user.name', self.committer_name])
-        if self.push_url:
-            repo.git(['remote', 'set-url', '--push', remote, self.push_url])
-
-        # Bump the Release label
-        df_path = self.workflow.builder.df_path
-        parser = DockerfileParser(df_path)
-        label_key = get_preferred_label_key(parser.labels, 'release')
-
-        attrs, key = self.find_current_release(parser, label_key)
-        next_release = self.get_next_release(attrs[key])
-        self.log.info("New Release: %s", next_release)
-        attrs[key] = next_release  # this modifies the file
-
-        # Stage it
-        repo.git(['add', os.path.basename(df_path)])
-
-        # Commit the change
-        repo.git(['commit',
-                  '--author={name} <{email}>'.format(name=self.author_name,
-                                                     email=self.author_email),
-                  '--message={message}'.format(message=self.commit_message)])
-
-        # Push it
-        self.log.info("Pushing to git repository")
-        repo.git(['push', remote])
-
-    def verify_branch(self, branch, branch_sha):
-        """
-        Raise exception if the branch is not at the correct commit.
-
-        :param branch: str, branch name
-        :param branch_sha: str, commit hash expected
-        """
-
-        if branch_sha != self.git_ref:
-            self.log.error("Branch '%s' is at commit %s (expected %s)",
-                           branch, branch_sha, self.git_ref)
-            raise RuntimeError("Not at expected commit")
-
-        self.log.info("Branch '%s' is at expected commit (%s)",
-                      branch, self.git_ref)
+        self.target = target
+        self.xmlrpc = koji.ClientSession(hub)
```

Comment:
It'll only attempt login if you provide a second parameter: https://github.com/projectatomic/atomic-reactor/blob/master/atomic_reactor/koji_util.py#L77